example/README.md

```diff
@@ -3,6 +3,67 @@
 Demonstrates how to use the cloud_firestore_mocks plugin.
 
 The example project comes from
-https://github.com/FirebaseExtended/flutterfire/tree/master/packages/cloud_firestore/example,
+https://github.com/FirebaseExtended/flutterfire/tree/master/packages/cloud_firestore/cloud_firestore/example,
 to which I've implemented two unit tests. See
 https://github.com/atn832/cloud_firestore_mocks/blob/master/example/test/widget_test.dart.
+
+
+# Driver Test: test_driver/cloud_firestore_behaviors
+
+The `test_driver/cloud_firestore_behaviors` driver test ensures the behavior of 
+cloud_firestore_mocks follows the real Firestore client.
+
+It runs the same set of assertions for the following three `Firestore` instances:
+
+- cloud_firestore backed by Cloud Firestore (project ID: flutter-firestore)
+- cloud_firestore backed by Firestore emulator
+- cloud_firestore_mocks
+
+## Start iOS Simulator
+
+Start iOS simulator. Driver tests require a simulator device to run.
+
+## Setup Firestore Emulator
+
+If you don't have `firebase` command, install [Firebase Cli](https://firebase.google.com/docs/cli#install-cli-mac-linux):
+
+```
+curl -sL https://firebase.tools | bash
+...
+~/Documents/cloud_firestore_mocks $ which firebase
+/usr/local/bin/firebase
+```
+
+Run Firestore emulator:
+
+```
+~/Documents/cloud_firestore_mocks $ firebase emulators:start --only firestore
+...
+i  firestore: For testing set FIRESTORE_EMULATOR_HOST=localhost:8080
+...
+```
+
+`test_driver/cloud_firestore_behaviors` assumes the emulator listen on port
+8080 (default) on localhost. This works for iOS simulator running in the same
```

Comment:
@atn832  I didn't setup anything special for my iOS simulator or Firestore emulator (or firebase.json).

# My environment
- Simulator version: Version 11.3.1 (SimulatorApp-912.5.1 SimulatorKit-570.3 CoreSimulator-681.17.2)
- My Mac OS Catalina: 10.15.3
- XCode 11.3.1
- firebase command 7.13.1

# Firestore emulator

In your emulator startup output, I don't see "Could not find config (firebase.json) so using defaults.". Aren't you using some firebase.json that defines security rules?

Mine:
```
~ $ firebase emulators:start --only firestore
⚠  Could not find config (firebase.json) so using defaults.
i  emulators: Starting emulators: firestore
⚠  No Firestore rules file specified in firebase.json, using default rules.
i  firestore: Serving ALL traffic (including WebChannel) on http://localhost:8080
⚠  firestore: Support for WebChannel on a separate port (8081) is DEPRECATED and will go away soon. Please use port above instead.
i  firestore: Emulator logging to firestore-debug.log
✔  firestore: Emulator started at http://localhost:8080
i  firestore: For testing set FIRESTORE_EMULATOR_HOST=localhost:8080
✔  All emulators started, it is now safe to connect.
```

Would you try running the emulator without firebase.json?

Reply:
Thanks for sharing the details of your environment. I had 7.0.0 installed. I must have had installed it long ago, and running the install script didn't actually update it. With 7.15.0, it works much better!

Version 7.0.0 required a firebase.json or it would fail (`Error: Not in a Firebase app directory (could not locate firebase.json)`). It also would fail if I hadn't pre-downloaded the emulator. But the new version downloads the newer emulator automatically when first run.

Reply:
Beautiful!

```
flutter: 00:00 +0: Firestore behavior comparison: Document creation by add (Cloud Firestore)
VMServiceFlutterDriver: Connected to Flutter application.
        path: satisfied (Path is satisfied), interface: en1
flutter: 00:01 +1: Firestore behavior comparison: Document creation by add (Firestore Emulator)
        path: satisfied (Path is satisfied), interface: en1
flutter: 00:01 +2: Firestore behavior comparison: Document creation by add (cloud_firestore_mocks)
flutter: 00:01 +3: Firestore behavior comparison: Document creation by setData (Cloud Firestore)
flutter: 00:02 +4: Firestore behavior comparison: Document creation by setData (Firestore Emulator)
flutter: 00:02 +5: Firestore behavior comparison: Document creation by setData (cloud_firestore_mocks)
flutter: 00:02 +6: Firestore behavior comparison: Timestamp field (Cloud Firestore)
flutter: 00:03 +7: Firestore behavior comparison: Timestamp field (Firestore Emulator)
flutter: 00:03 +8: Firestore behavior comparison: Timestamp field (cloud_firestore_mocks)
flutter: 00:03 +9: Firestore behavior comparison: Unsaved documens (Cloud Firestore)
flutter: 00:04 +10: Firestore behavior comparison: Unsaved documens (Firestore Emulator)
flutter: 00:04 +11: Firestore behavior comparison: Unsaved documens (cloud_firestore_mocks)
flutter: 00:04 +12: Firestore behavior comparison: Nested objects creation with updateData (Cloud Firestore)
flutter: 00:05 +13: Firestore behavior comparison: Nested objects creation with updateData (Firestore Emulator)
flutter: 00:05 +14: Firestore behavior comparison: Nested objects creation with updateData (cloud_firestore_mocks)
flutter: 00:05 +15: Firestore behavior comparison: Nested objects update (Cloud Firestore)
flutter: 00:06 +16: Firestore behavior comparison: Nested objects update (Firestore Emulator)
flutter: 00:06 +17: Firestore behavior comparison: Nested objects update (cloud_firestore_mocks)
flutter: 00:06 +18: (tearDownAll)
flutter: 00:06 +19: All tests passed!
Stopping application instance.
```

Reply:
Glad to hear it worked.